qiskit_nature/operators/second_quantization/quadratic_hamiltonian.py

```diff
@@ -0,0 +1,306 @@
+# that they have been altered from the originals.
+
+"""The QuadraticHamiltonian class."""
+
+from typing import Optional, Tuple
+
+import numpy as np
+import scipy.linalg
+from qiskit.quantum_info.operators.mixins import TolerancesMixin
+from qiskit_nature.operators.second_quantization.fermionic_op import FermionicOp
+
+
+class QuadraticHamiltonian(TolerancesMixin):
+    r"""A Hamiltonian that is quadratic in the fermionic ladder operators.
+
+    A quadratic Hamiltonian is an operator of the form
+
+    .. math::
+        \sum_{p, q} M_{pq} a^\dagger_p a_q
+        + \frac12 \sum_{p, q}
+            (\Delta_{pq} a^\dagger_p a^\dagger_q + \text{h.c.})
+        + \text{constant}
+
+    where :math:`M` is a Hermitian matrix and :math:`\Delta` is an antisymmetric matrix.
+    """
+
+    def __init__(
+        self,
+        hermitian_part: Optional[np.ndarray] = None,
+        antisymmetric_part: Optional[np.ndarray] = None,
+        constant: float = 0.0,
+    ) -> None:
+        r"""Initialize a QuadraticHamiltonian.
+
+        Args:
+            hermitian_part: The matrix :math:`M` containing the
+                coefficients of the terms that conserve particle number.
+            antisymmetric_part: The matrix :math:`\Delta` containing the
+                coefficients of the terms that do not conserve particle number.
+            constant: An additive constant term.
+
+        Raises:
+            ValueError: Either a Hermitian part or antisymmetric part must be specified.
+        """
+        if hermitian_part is not None:
+            self._n_orbitals, _ = hermitian_part.shape
+        elif antisymmetric_part is not None:
+            self._n_orbitals, _ = antisymmetric_part.shape
+        else:
+            raise ValueError("Either a Hermitian part or antisymmetric part must be specified.")
+
+        # TODO maybe validate matrix properties
+        self._hermitian_part = hermitian_part
+        self._antisymmetric_part = antisymmetric_part
+        self._constant = constant
+
+        if self._hermitian_part is None:
+            self._hermitian_part = np.zeros((self._n_orbitals, self._n_orbitals))
+        if self._antisymmetric_part is None:
+            self._antisymmetric_part = np.zeros((self._n_orbitals, self._n_orbitals))
+
+    @property
+    def hermitian_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that conserve particle number."""
+        return self._hermitian_part
+
+    @property
+    def antisymmetric_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that do not conserve particle number."""
+        return self._antisymmetric_part
+
+    @property
+    def constant(self) -> float:
+        """The constant."""
+        return self._constant
+
+    def _fermionic_op(self) -> FermionicOp:
+        """Convert to FermionicOp."""
+        terms = []
+        # TODO I shouldn't have to use string labels
```

Comment:
You can refactor this now that #506 is merged.

Reply:
Done.